jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -387,10 +387,33 @@ def run(self):
             directive_key
         )
 
-        button_text = self.options.pop("button_text", "Try it with Jupyterlite!")
+        # Use button_text from try_examples_global_button_text if not provided
+        if self.env.config.try_examples_global_button_text is not None:
+            button_text = self.options.get(
+                "button_text", self.env.config.try_examples_global_button_text
+            )
+        else:
+            button_text = self.options.pop("button_text", "Try it with JupyterLite!")
```

Comment:
```suggestion
        default_button_text = self.env.config.try_examples_global_button_text
        if default_button_text  is None:
            default_button_text = "Try it with JupyterLite!"
        button_text = self.options.pop("button_text", default_button_text)
```

Two things. The use of `pop` instead of `get` is essential. We don't want "button_text" to remain in the `self.options` dictionary. The remaining key value pairs will passed as options into a url later on.

Second, I think the way I've suggested is cleaner because the control flow logic is used explicitly used to determine the default button text, and we avoid duplicating the logic to pop something from the dictionary.

Reply:
Thanks! I had an issue come up where the button disappeared completely when I had used `pop` to display a global warning, I'm not sure why – so I had replaced everything with `get` instead.

I'm not sure why that happened, let me try to debug that.

Reply:
Your logic was much better, and everything works as expected – tested all of them again. Addressed in 47621ff2dc743ca4491046bc97b9c04009e1291e.